pkg/storage/stores/indexshipper/compactor/retention/util_test.go

```diff
@@ -121,10 +121,11 @@ type table struct {
 	chunks map[string][]chunk.Chunk
 }
 
-func (t *table) ForEachChunk(callback ChunkEntryCallback) error {
+func (t *table) ForEachChunk(ctx context.Context, callback ChunkEntryCallback) error {
 	for userID, chks := range t.chunks {
 		i := 0
-		for _, chk := range chks {
+		for j := 0; j < len(chks) && ctx.Err() == nil; j++ {
```

Comment:
It's not ideal to change this in the test. The actual implementations include tests and this ensures that we can test the behavior around the timeout in `retention.go`